source/extensions/filters/http/jwt_authn/jwks_cache.cc

```diff
@@ -63,6 +66,17 @@ class JwksDataImpl : public JwksCache::JwksData, public Logger::Loggable<Logger:
     return setKey(std::move(jwks), getRemoteJwksExpirationTime());
   }
 
+  const std::unique_ptr<TokenCache>& getTokenCache() override {
```

Comment:
Do we need to return unique_ptr?  If you are not passing the ownership, you don't need to return unique_ptr. 
It seems that you are not c++ programmer.